packages/bitcore-node/src/modules/firocoin/api/search.ts

```diff
@@ -0,0 +1,104 @@
+import express = require('express');
+import { TransactionStorage } from '../../../models/transaction';
+import { ContractStorage } from '../models/contract';
+import { TokenStorage } from '../models/token';
+const router = express.Router({ mergeParams: true });
+
+router.get('/:content', async (req, res) => {
+  let { chain, network, content } = req.params;
+  try {
+    content = Buffer.from(content, 'base64').toString('utf-8');
+
+    // TODO: Other Address
+    // const regexBitcoinAddresses = /([13]|bc1)[A-HJ-NP-Za-km-z1-9]{27,34}/g;
+
+    const regexNativeAddress = /T[A-HJ-NP-Za-km-z1-9]{33}/g;
+    const matchNativeAddress = content.match(regexNativeAddress);
+    if (matchNativeAddress && content.length === 34) {
+      const nativeAddress = matchNativeAddress[0];
+      res.json({
+        type: 'account',
+        native: nativeAddress,
+      });
+      return;
+    }
+
+    const regexHexAddress = /[0-9a-fA-F]{40}/g;
+    const matchHexAddress = content.match(regexHexAddress);
+    if (matchHexAddress && content.length === 40) {
+      const hexAddress = matchHexAddress[0];
+      const contract = await ContractStorage.collection.findOne({ chain, network, contractAddress: hexAddress });
+      if (contract) {
+        res.json({
+          type: 'contract',
+          address: hexAddress,
+        });
+      } else {
+        res.json({
+          type: 'account',
+          hex: hexAddress,
+        });
+      }
+      return;
+    }
+
+    const regexBlockNumber = /[0-9]{1,15}/g;
+    const matchBlockNumber = content.match(regexBlockNumber);
+    if (matchBlockNumber && content.length <= 15) {
+      const blockId = matchBlockNumber[0];
```

Comment:
block ids should be hash instead of number, since we could have multiple block hashes with the same block number.